hicommonwealth/commonwealth
# Show join community banner when user has not joined

## Link to Issue
Closes: #4380 

## Description of Changes
- Created new `Banner` component but [per Zak's request (point 1 in linked comment)](https://github.com/hicommonwealth/commonwealth/issues/4380#issuecomment-1621806681) it handles only one variation that is needed in this particular story. This means that [every variation from the design](https://www.figma.com/file/eIVp33a1oCu0AtcLwSbGjr/%F0%9F%9A%A7-Components-and-Patterns?type=design&node-id=789%3A10883&mode=design&t=HyobTlUa6qSug8qe-1) should be implemented in [separate ticket](https://github.com/hicommonwealth/commonwealth/issues/4407) along with storybook. 
- Renamed old `CWBanner` to `Old_CWBanner` to avoid confusions
- Previously, the whole logic related to joining community was situated in `LoginSelector` component. Because we needed this button to be available in 3 places (login selector in topbar, create thread page, view thread page), I abstracted out the logic to reusable hook `useJoinCommunity`. The logic was not changed in any way (it was copied) so the behaviour should be unchanged.
- created `JoinCommunityBanner` that is used in two places and `useJoinCommunityBanner` hook that handles the logic of showing/closing the banner within 1 week period of time.
- created `useUserActiveAccount` to make sure that the information if user has joined community is available (and sync) all over the components that need this info 


https://github.com/hicommonwealth/commonwealth/assets/14819225/383f2897-1640-4334-bd47-5a6c0eefff49


## Test Plan
  - while not logged in
    - view thread page
      - editor and banner should not be visible 
    - create new thread page  
      - page is not accessible  
  - while logged in but not joined the community
    - view thread page
      - comment editor visible but should be disabled
      - banner should be visible
      - banner can be closed by clicking X 
      - banner should not be visible on page reload (unless 1 week passed)
      - banner should be visible again when local storage is cleaned 
      - clicking join button in banner should result in joining the community
    - create new thread page  
      - post editor visible but should be disabled
      - banner should be visible
      - banner can be closed by clicking X 
      - banner should not be visible on page reload (unless 1 week passed)
      - banner should be visible again when local storage is cleaned 
      - clicking join button in banner should result in joining the community
    - while logged in and joined the community
      - banner should not be visible anywhere 

## Other Considerations

- there are other tickets in the bucket that will extend the functionality of this PR #4306
